.github/workflows/lint.yaml

```diff
@@ -36,8 +36,6 @@ jobs:
         run: |
           python -m pip install pre-commit
       - name: Run pre-commit checks
-        # Continue on error to display the formatting diff in case lint fails.
-        continue-on-error: true
```

Comment:
This is what prevented the CI to be red when the pre-commit job failed.

We'll have to find a better way to display the formatting diff.